src/Google-Plus/Provider.php

```diff
@@ -66,10 +66,11 @@ protected function getUserByToken($token)
     protected function mapUserToObject(array $user)
     {
         return (new User())->setRaw($user)->map([
-            'id'     => $user['id'], 'nickname' => Arr::get($user, 'nickname'),
-            'name'   => $user['displayName'],
-            'email'  => $user['emails'][0]['value'],
-            'avatar' => Arr::get($user, 'image.url'),
+            'id' => $user['names'][0]['metadata']['source']['id'],
+            'nickname' => $user['names'][0]['displayName'],
+            'name' => $user['names'][0]['displayName'],
+            'email' => $user['emailAddresses'][0]['value'],
+            'avatar' => $user['photos'][0]['url'],
```

Comment:
Is the first photo always the avatar? What happens if the user has no avatar?

Reply:
If not set any ava Google uses an image with your name first char as the avatar. I removed my ava and  Google automatically set this ava for me https://lh6.googleusercontent.com/-IPi1e97NLh4/AAAAAAAAAAI/AAAAAAAAAAA/ACevoQNXgMp-CZ4U_4r_aIDPCcDJil4N_g/s100/photo.jpg